pulsar-client-cpp/lib/ExecutorService.cc

```diff
@@ -55,9 +55,14 @@ DeadlineTimerPtr ExecutorService::createDeadlineTimer() {
 }
 
 void ExecutorService::close() {
-    io_service_.stop();
-    work_.reset();
-    worker_.join();
+    // Ensure this service has not already been closed. This is
+    // because worker_.join() is not re-entrant on Windows
+    if (work_)
+    {
```

Comment:
Even if Windows related, can you submit this fix as a separate PR so that it gets more visible? 

Also, the `{` on new line will fail on the `make check-format`. You should be able to do `make format` and automatically fix all the formatting (that uses `clang-format` underneath)

Reply:
Ok, I took this change out of this PR and into a new one, #4084 

In addition, I realized that I didn't notice the check-format failure because the Windows version of the CMake files were omitting those targets. I re-added them and was able to correct a host of formatting mistakes.